identity-admin/webapp/components/search/SearchInput.tsx

```diff
@@ -0,0 +1,56 @@
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { buildSearchUrl } from '../../utils/search-util';
+
+const SearchInput = (): JSX.Element => {
+  const router = useRouter();
+  const { status, name, email } = router.query;
+  const [nameValue, setNameValue] = useState<string>(
+    typeof name === 'string' ? name : ''
+  );
```

Comment:
I see what you're doing here, and there's an easier way to do this.

Web forms keep their own internal state. That's not a React thing, it's a standard browser thing. You store a value in a form's state keyed to the input's `name` attribute, then you can access the form state inside your `onSubmit`.

Inputs can also take a `defaultValue` attribute, which could come straight from the props.

```tsx
export const Untitled = (props: { firstName: string }) => {
  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    console.log(event.currentTarget.firstName.value);
    // -> "Rob"
  };

  return (
    <form onSubmit={handleSubmit}>
      <input name="firstName" defaultValue={props.firstName} />
      <input type="submit" value="Submit" />
    </form>
  );
};
```

So you don't need to keep `nameValue` or `emailValue` in React state here.